usecases/linearFunction/test_optimize.py

```diff
@@ -34,33 +37,48 @@ def __init__(self, yaml_file_metadata, yaml_file_data):
             data_df = np.asarray(data[idx]['df'])
             assert(a == data[idx]['a'])
             single_model_error = LinearModelError(x_function, x_derivative, data_f, data_df, a)
             # read in the parameters that are given in the experiment (in this case constant offset a)
             parameter = single_model_error.get_parameter_dict()
-            # add parameters that are model parameters (not given in the experimental data)
-            parameter.define("b")
+            # store the parameter list here to access it later, e.g. in the evaluation of the likelihood
+            self.linearModelParameters.append(parameter)
+            # add the parameter list with the model error to the joint latent list
             self.add(single_model_error, parameter)
 
-        # define one shared latent parameter (only one parameter b for all linear models)
-        self.latent.add_by_name('b')
+    def latent_as_tensor_variable(self, model, prior_pymc3):
+        with model:
+            #array = [prior_pymc3[key] for key, var in prior_pymc3.items()]
+            array = []
+            for latent_parameters in self.latent._index_mapping:
+                # latent_parameters is now a dict of all parameters with key related to the model key and value being the name of the variable in this list
+                # get value of first iterator, which is currently supposed to be all identical
+                # TODO we need to define for each latent variable a new name that is then used to identify the prior, or set the initial value
+                var_name = next(iter(latent_parameters.values()))
+                array.append(prior_pymc3[var_name])
+            return tt.as_tensor_variable(array)
 
 
 class TestOptimize(unittest.TestCase):
-    def test_linear_virtual_data(self):
+    def test_linear(self):
         virtual_experiment_metadata_file = \
-            Path(__file__).parent / 'virtual_experiment_linear_model_meta.yaml'
+            Path(__file__).parent / 'virtual_experiment_linear_meta.yaml'
         with open(virtual_experiment_metadata_file, "r") as f:
             d = yaml.load(f, Loader=yaml.FullLoader)
 
         all_experiments_model_error = \
-            MultiLinearModelError('virtual_experiment_linear_model_meta.yaml',
-                                  'virtual_experiment_linear_model_data.yaml')
-
-        start_vector = np.array([0.7])
+            MultiLinearModelError('virtual_experiment_linear_meta.yaml',
+                                  'virtual_experiment_linear_data.yaml')
+        # define one shared latent parameter (only one parameter b for all linear models)
+        # add all variables 'b' in all model parameter lists as a shared latent variable
+        # return index in the global vector
+        index_b = all_experiments_model_error.latent.add_by_name('b')
+        start_vector = np.empty([len(all_experiments_model_error.latent)])
+        start_vector[index_b] = 0.7
```

Comment:
Thank you for trying out my recommendation. But actually, I must admit that this also looks a bit clumsy. And it only makes sense if the "add_by_name" really returns a zero. I mean, if the _user_ must initialize the `start_vector` -- instead of just telling the library "Set the value of my latent parameter to 0.7" -- the "index_b" is rather pointless.

So ideally, we should IMO generalize the idea of this "UncorrelatedNormalPrior" (that takes care of setting the latent parameters _and_ the prior and, thus, hides the internal ordering from the user (if he wishes)) to any 'prior' (including pymc3, just numbers, MVNs,...)

Reply:
I would say that this strongly related to the question, if we should have an internal (related to the total latent) vector representation that we can access and extract as a vector (e.g. to be extracted and used [here](https://github.com/BAMresearch/ModelCalibration/blob/a6c52122b82b3499243d19fe654f8f67d44e3c70/usecases/linearFunction/test_optimize.py#L82.) Currently we do not have this (e.g. each entry in the latent vector might be related to different entries in the local vectors), and they are also just numbers - as opposed to e.g. a distribution (even a multivariate distribution might be required). As for the return of zero for add_by_name, I would say this is also works for more than one variable, since if your latent vector has 5 entries, each index would return the corresponding index that the variable is related to, and then you can directly access this entry in the latent variable via the index.